### PR TITLE
creates working build on linux

### PR DESCRIPTION
**Pull request by luxe**
project builds in linux now.  
It's a custom makefile, that should probably be turned into cmake sometime.  
Hopefully it still builds under windows too.

### CODE REVIEW

File: Mudd/MuddCommTest.cpp [hunk 1]
```diff
@@ -58,7 +58,8 @@ void StartClient()
         {
             if (line != "")
             {
-                msgs.Push(CommandDecoder::Decode(line));
+                auto msg = CommandDecoder::Decode(line);
```
Comment (luxe) [inline Mudd/MuddCommTest.cpp]: couldn't pass a temporary unique_ptr into a function taking a unique_ptr&.  
So I just made a local copy before passing it in.  Only way I could get it to compile

File: Mudd/stdafx.h [hunk 2]
```diff
@@ -8,7 +8,10 @@
 #include "targetver.h"
 
 #include <stdio.h>
+
+#if WIN32 || WIN64
 #include <tchar.h>
+#endif
```
Comment (luxe) [inline Mudd/stdafx.h]: ¯\_(ツ)_/¯ good enough for now

**Comment by gheckman** [comment]
I'll check this when I get home to make sure it still builds in Windows.

**Comment by luxe** [comment]
k

**Comment by gheckman** [comment]
I'm ACTUALLY looking into it now. A month later.
I'm a piece of crap.

**Pull request merged**